spring-integration-core/src/main/java/org/springframework/integration/store/AbstractKeyValueMessageStore.java

```diff
@@ -294,6 +295,15 @@ public void removeMessageGroup(Object groupId) {
 		}
 	}
 
+	@Override
+	public void setGroupCondition(Object groupId, String condition) {
+		MessageGroupMetadata metadata = getGroupMetadata(groupId);
+		if (metadata != null) {
```

Comment:
Perhaps we need an assertion here, to ensure `setGroupCondition` is only called after the current record has been added to the group (and metadata will be present).

Reply:
I don't think it is relevant. It is only the fact of an aggregator that we add a record then calculate a condition.
In other cases it might be set independently and unconditionally.

Reply:
ok